tests/unit/test_utils.py

```diff
@@ -647,21 +645,56 @@ def test_split_auth_from_netloc(netloc, expected):
 
 
 @pytest.mark.parametrize('auth_url, expected_url', [
-    ('https://user:pass@domain.tld/project/tags/v0.2',
-     'https://domain.tld/project/tags/v0.2'),
     ('https://domain.tld/project/tags/v0.2',
      'https://domain.tld/project/tags/v0.2',),
-    ('https://user:pass@domain.tld/svn/project/trunk@8181',
-     'https://domain.tld/svn/project/trunk@8181'),
     ('https://domain.tld/project/trunk@8181',
      'https://domain.tld/project/trunk@8181',),
+    ('https://user:pass@domain.tld/project/tags/v0.2',
+     'https://domain.tld/project/tags/v0.2'),
+    ('https://user:pass@domain.tld/svn/project/trunk@8181',
+     'https://domain.tld/svn/project/trunk@8181'),
     ('git+https://pypi.org/something',
      'git+https://pypi.org/something'),
-    ('git+https://user:pass@pypi.org/something',
-     'git+https://pypi.org/something'),
     ('git+ssh://git@pypi.org/something',
      'git+ssh://pypi.org/something'),
+    ('git+https://user:pass@pypi.org/something',
+     'git+https://pypi.org/something'),
 ])
 def test_remove_auth_from_url(auth_url, expected_url):
     url = remove_auth_from_url(auth_url)
     assert url == expected_url
+
+
+@pytest.mark.parametrize('auth_url, expected_url', [
+    ('http://domain.tld:8080/',
+     'http://domain.tld:8080/',),
+    ('http://user@domain.tld:8080/',
+     'http://user@domain.tld:8080/',),
+    ('https://domain.tld/project/tags/v0.2',
+     'https://domain.tld/project/tags/v0.2',),
+    ('https://domain.tld/project/trunk@8181',
+     'https://domain.tld/project/trunk@8181',),
+    ('https://user:pass@domain.tld/project/tags/v0.2',
+     'https://user:****@domain.tld/project/tags/v0.2'),
+    ('https://user:pass@domain.tld/svn/project/trunk@8181',
+     'https://user:****@domain.tld/svn/project/trunk@8181'),
+    ('https://user:pass:word@domain.tld/',
+     'https://user:****@domain.tld/',),
+    ('https://user:pass%3Aword@domain.tld/',
+     'https://user:****@domain.tld/',),
+    ('https://us%3Aer:pass@domain.tld/',
+     'https://us%3Aer:****@domain.tld/',),
+    ('https://user:pass@word@domain.tld/',
+     'https://user:****@domain.tld/',),
+    ('https://user:@domain.tld/',
+     'https://user:****@domain.tld/',),
+    ('git+https://pypi.org/something',
+     'git+https://pypi.org/something'),
+    ('git+ssh://git@pypi.org/something',
+     'git+ssh://git@pypi.org/something'),
+    ('git+https://user:pass@pypi.org/something',
+     'git+https://user:****@pypi.org/something'),
```

Comment:
This seems like more test cases than you need. If you break out `redact_netloc()` like I suggested above and test it separately, you should be able to reduce the number of test cases you need for combinatorial reasons. You can focus on testing `redact_netloc()` well (which is a simpler function). And then `test_redact_password_from_url()` can be limited to making sure that `redact_password_from_url()` is "wired up" correctly from `redact_netloc()` and `transform_url()`.